### PR TITLE
Add Spanish translation #1684

### PR DESCRIPTION
**Pull request by macagua**
Add Spanish translation for this addon

### CODE REVIEW

**Comment by macagua** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+msgid "Adds sample content types for testing"
+msgstr "Añadir tipos de contenido de ejemplo para pruebas"
+
+#: plone/restapi/configure.zcml:65
+msgid "Adds sample workflows for testing"
+msgstr "Añadir flujos de trabajo de muestra para realizar pruebas"
+
+#: plone/restapi/services/aliases/add.py:106
+msgid "Alternative urls that point to themselves will cause an endless cycle of redirects."
+msgstr "Las URL alternativas que apuntan a sí mismas provocarán un ciclo interminable de redireccionamientos."
```
Comment (erral) [inline src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:45]: no tengo claro si se debería usar ciclo o bucle

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+msgid "Enables Volto Blocks support"
+msgstr "Habilitar soporte a Bloques Volto"
+
+#: plone/restapi/configure.zcml:81
+msgid "Enables blocks on the Document content type"
+msgstr "Habilitar bloques en el tipo de contenido Documento"
+
+#: plone/restapi/services/contextnavigation/get.py:133
+msgid "Enter a valid scale name (see 'Image Handling' control panel) to override (e.g. icon, tile, thumb, mini, preview, ... ). Leave empty to use default (see 'Site' control panel)."
+msgstr "Ingrese un nombre de escala válido (consulte 'Manipulación de imágenes' en el panel de control) para anular (por ejemplo, icon, tile, thumb, mini, preview...). Déjelo en blanco para usar el valor predeterminado (consulte 'Sitio' en el panel de control)."
```
Comment (erral) [inline src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:69]: Ingresar - > introducir

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+msgid "If enabled, the portlet will not show document type icons."
+msgstr "Si está habilitado, el portlet no mostrará los iconos de tipo de contenido."
+
+#: plone/restapi/services/contextnavigation/get.py:145
+msgid "If enabled, the portlet will not show thumbs."
+msgstr "Si está habilitado, el portlet no mostrará miniaturas."
+
+#: plone/restapi/services/users/add.py:296
+msgid "If you pass 'old_password' you have to pass 'new_password'"
+msgstr "Si ingresa 'Contraseña anterior', debe ingresar 'Contraseña nueva'"
```
Comment (erral) [inline src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:90]: ingresar -> introducir

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+#: plone/restapi/services/users/add.py:290
+msgid "If you pass 'reset_token' you have to pass 'new_password'"
+msgstr "Si ingresa 'Restablecer token' debe ingresar 'Nueva contraseña'"
```
Comment (erral) [inline src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:94]: ingresar  -> introducir

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+msgid "Layout"
+msgstr "Diseño"
+
+#: plone/restapi/services/contextnavigation/get.py:195
+msgid "Navigation"
+msgstr "Navegación"
+
+#: plone/restapi/services/contextnavigation/get.py:132
+msgid "Override thumb scale"
+msgstr "Anular escala de miniaturas"
```
Comment (erral) [inline src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:106]: sobreescribir?

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+msgid "RESTful hypermedia API for Plone - Uninstall"
+msgstr "API hipermedia RESTful para Plone - Desinstalar"
+
+#: plone/restapi/configure.zcml:47
+msgid "RESTful hypermedia API for Plone."
+msgstr "API hipermedia RESTful para Plone."
+
+#: plone/restapi/services/history/patch.py:28
+msgid "Reverted to revision ${version}"
+msgstr "Revertido para revisión ${version}"
```
Comment (erral) [inline src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:126]: Revertido a revisión  no?

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+msgid "You are not authorized to perform this action"
+msgstr "No está autorizado a realizar esta acción."
+
+#: plone/restapi/services/email_send/post.py:91
+msgid "You are receiving this mail because ${sender_fullname} sent this message via the site ${portal_title}:"
+msgstr "Está recibiendo este correo porque ${sender_fullname} envió este mensaje a través del sitio ${portal_title}:"
+
+#: plone/restapi/services/users/add.py:114
+msgid "You can't send both password and sendPasswordReset."
+msgstr "No puede ingresar la contraseña y 'Enviar un correo electrónico de confirmación con un enlace para establecer la contraseña'."
```
Comment (erral) [inline src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:166]: aquí ingresar1debería ser enviar

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+
+#: plone/restapi/services/users/add.py:322
+msgid "You can't set a password without a password reset token."
+msgstr "No puede establecer una contraseña sin un token de restablecimiento de contraseña."
+
+#: plone/restapi/services/users/update.py:125
+msgid "You can't update the properties of this user"
+msgstr "No puedes actualizar las propiedades de este usuario."
```
Comment (erral) [inline src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:174]: No puede. Utilizamos el usted.

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+
+#: plone/restapi/services/users/add.py:284
+msgid "You can't use 'reset_token' and 'old_password' together."
+msgstr "No puede utilizar 'Restablecer token' y 'Contraseña anterior' juntas."
+
+#: plone/restapi/services/users/add.py:109
+msgid "You have to either send a password or sendPasswordReset."
+msgstr "Debe ingresar una contraseña o 'Enviar un correo electrónico de confirmación con un enlace para establecer la contraseña'."
```
Comment (erral) [inline src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:182]: aquí ingresar es enviar

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+
+#: plone/restapi/services/users/add.py:154
+msgid "You need AddPortalMember permission."
+msgstr "Necesita el permiso AddPortalMember."
+
+#: plone/restapi/services/users/add.py:329
+msgid "You need to be logged in as the user '${username}' to set the password."
+msgstr "Debe iniciar sesión como usuario '${username}' para establecer la contraseña."
+
```
Comment (erral) [inline src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:191]: igual 'para poder establecer' quedaría mejor?

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po [hunk 11]
```diff
@@ -0,0 +1,279 @@
...
+msgstr "Dependencia faltante"
+
+#. Default: "If selected, the navigation tree will only show the current folder and its children at all times."
+#: plone/restapi/services/contextnavigation/get.py:84
+msgid "help_current_folder_only"
+msgstr "Si se selecciona, el árbol de navegación siempre mostrará solo la carpeta actual y sus hijos."
+
+#. Default: "Whether or not to show the top, or 'root', node in the navigation tree. This is affected by the 'Start level' setting."
+#: plone/restapi/services/contextnavigation/get.py:69
+msgid "help_include_top_node"
```
Comment (erral) [inline src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:204]: @davisagli I don't know why do we have these strings here... I think they are not shown anywhere in the API... we should review why we are passing them from gettext

**Review by erral — CHANGES_REQUESTED** [review]

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po (hunk 1 above)
Comment (macagua) [reply src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po]: Hecho con commit [4b907ad2]

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po (hunk 2 above)
Comment (macagua) [reply src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po]: Hecho con commit [4b907ad2]

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po [hunk 12]
```diff
@@ -0,0 +1,279 @@
...
+msgstr "Ingrese un nombre de escala válido (consulte 'Manipulación de imágenes' en el panel de control) para anular (por ejemplo, icon, tile, thumb, mini, preview...). Déjelo en blanco para usar el valor predeterminado (consulte 'Sitio' en el panel de control)."
+
+#: plone/restapi/services/users/add.py:163
+msgid "Error in fields. ${errors_to_string}"
+msgstr "Errores en los campos. ${errors_to_string}"
+
+#. Default: "The reset_token is expired."
+#: plone/restapi/services/users/add.py:312
+msgid "Expired Token"
+msgstr "Token Venció"
```
[the remark this answers is not in the document]
Comment (macagua) [reply src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po]: Hecho con commit [4b907ad2]

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po (hunk 3 above)
Comment (macagua) [reply src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po]: Hecho con commit [4b907ad2]

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po (hunk 4 above)
Comment (macagua) [reply src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po]: Hecho con commit [4b907ad2]

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po (hunk 7 above)
Comment (macagua) [reply src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po]: Hecho con commit [8589c58]

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po (hunk 8 above)
Comment (macagua) [reply src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po]: Hecho con commit [8589c58]

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po (hunk 9 above)
Comment (macagua) [reply src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po]: Hecho con commit [8589c58]

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po (hunk 10 above)
Comment (macagua) [reply src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:191]: Hecho con commit [8589c58]

File: src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po (hunk 11 above)
Comment (davisagli) [reply src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:204]: @erral Good point. It looks like it's a schema that was copied from plone.app.portlets. The schema is used to deserialize options from the request but its fields don't need translations.

Comment (erral) [reply src/plone/restapi/locales/es/LC_MESSAGES/plone.restapi.po:204]: We can handle that in a separate PR, I will try to prepare it. Thanks!

**Comment by macagua** [comment]
@erral Only I made a ``git pull origin master`` and ``git rebase master`` at ``spanish-translation`` branch

> Why do we have so many changes here? It looks like the PR is not properly rebased, right?

@erral Only I made a ``git pull origin master`` and ``git rebase master`` at ``spanish-translation`` branch

**Review by erral — APPROVED** [review]

File: CHANGES.rst [hunk 13]
```diff
@@ -14,6 +14,7 @@ Changelog
 New features:
 
 
+- Add Spanish translation. [macagua] (#1684)
```
Comment (davisagli) [inline CHANGES.rst:17]: @macagua Please read the note at the top of this file -- changelog entries should be added in the news folder, not here. This file is only updated at the time that we make a release.

(Your change is obviously not part of the 8.43.0 release, which was already released!)

**Review by davisagli — CHANGES_REQUESTED** [review]

File: CHANGES.rst (hunk 13 above)
Comment (macagua) [reply CHANGES.rst]: > @macagua Please read the note at the top of this file -- changelog entries should be added in the news folder, not here. This file is only updated at the time that we make a release.
> 
> (Your change is obviously not part of the 8.43.0 release, which was already released!)

@davisagli done with commits [0a94255](https://github.com/plone/plone.restapi/pull/1685/commits/0a94255611b1b11252234471db84cb6534704f37) and [53207ec](https://github.com/plone/plone.restapi/pull/1685/commits/53207ec99e396a21cb5662d0ba07baea8e7e6fb1)

File: news/1684.feature [hunk 14]
```diff
@@ -0,0 +1 @@
+Add Spanish translation (#1684) @macagua
```
Comment (davisagli) [inline news/1684.feature:1]:
```suggestion
Add Spanish translation @macagua
```

(towncrier adds the link to the PR when it updates CHANGES.rst)

**Review by davisagli — APPROVED** [review]

**Pull request merged**